### ISSUE TITLE
Installations not being removed from debug on Takistan after being destroyed

### ISSUE THREAD

**Issue by HeroesandvillainsOS** [ALiVEOS/ALiVE.OS#316]
I just noticed that although installations are set up and seem to be functioning just fine on Takistan (they have enemies, objects (for the most part), recruit civs, etc), almost all of them won't remove from debug even after properly being destroyed.

Here is a test mission that demonstrates the issue pretty well. It's with vanilla factions and about as bare bones as it can get.

https://www.dropbox.com/s/uh3wh6ug5kvwa2q/CUP_test.Takistan.pbo?dl=0

Requires CBA, ALiVE and CUP Terrains

**Comment by highhead** [comment]
yo mate, are you using preplaced installations on the AI commander module?

**Comment by highhead** [comment]
PS: didnt look at the mission yet. But you should be able to replicate it in vanilla as it sounds.

**Comment by HeroesandvillainsOS** [comment]
For this example, yes. I think I set it to medium or high so one of you guys could get into the meat and bones of the ticket pretty quickly.

That said, I first noticed it in a mission that had them set to off. So on Takistan, at least, the pre-placed installation setting doesn't seem to make a difference.

I tried the exact setup from the linked mission above on Stratis, and couldn't find any examples of installations not being removed from debug. But to be fair, I only tried maybe 5 or 6 on Stratis before deeming it working properly. But on Takistan it's like the opposite, where you need to blow up 5 or 6 and maybe eventually 1 of them will be removed. It's weird.

If it would be helpful, I can run another test if you can give me some testing parameters.

**Comment by highhead** [comment]
Hey!

It can take up to a minute until next analysis happens and then the markers will be removed. I just verified its working on Takistan and no single marker wasn't removed after a while. You can override it with Admin/Toggle OPCOM installations, to check it quicker too. I blew up several Recruitment centers, Weapon depots and IED factories by using "cursortarget setdamage 1" when looking at the buildings.

Can you please check if its not only taking a while just to be sure!
Thanks

**Comment by highhead** [comment]
Alternate reply:

Awright geeezzaa! It can take up ter a minute until next analysis 'appens an' then da markers'll be removed. I just verified its workin' on Takistan an' no single marker wasn't removed after a while. Thee can override i' wiv Admin/Toggle OPCOM installashuns, ter check i' quicker an' all. I blew up several Recruitmen' cen'ers, Weapon depots an' IED factories by usin' "cursortarget setdamage 1" when lookin' at da buildings. Sorted mate.

**Comment by HeroesandvillainsOS** [comment]
The alternate reply was what I was after. :)

So I should have noted this in the first post (my bad), but toggling the OPCOM installations in the Admin options is exactly how I've been determining whether or not installations has been successfully removed. And to test, I've been using actual explosive charges and not any kind of description model code (some Takistan buildings take a bunch of charges to destroy, as we've discussed before, upwards of 4+ explosives in places)

So in my Stratis test, I destroyed one building at a time, several times, and all of them would be removed from the debug toggle instantaneously.

But with the same exact setup on Takistan, almost none of them would be immediately removed from the installations debug toggle, and would continue to show up in the debug toggle.

1. How long should I wait to make sure they are being removed as intended? 

2. Any guesses why on Stratis they get immediately removed in debug when destroyed, but not on Takistan? I'm thinking it may have to do with how much damage these buildings can take. I have to wonder if your test would have been as successful if you had used actual explosives.

Anyway, I'll check later and wait it out a bit longer and see if they go away from debug at some point.

**Comment by highhead** [comment]
they also vanish for me instantly with debug toggle on takistan actually. can you try cursortarget setdamage 1 when looking at the building (stand back :)

**Comment by HeroesandvillainsOS** [comment]
Yeah? With the mission in the first post they removed immediately in the debug toggle for you? Or was it with your own test mission?

**Comment by highhead** [comment]
Yeah! It just works for me! Without a hitch (though I wouldn't mind doing some optimizations on selecting correct buildings for some types (exlude small bunkers f.e.).

**Comment by highhead** [comment]
just keep in mind to let the debug toggle fade out completely and when you call it again the building markers will be gone.

**Comment by HeroesandvillainsOS** [comment]
Ok. I think I see what's going on here (fo shizzle :) 

It seems to me, several CUP building models can look to the naked eye as being completely destroyed. But...

Using cursortarget setdamage 1, its now clear to me some "buildings" comprise of several models all in one, and if the player doesn't use enough explosives, or has poor explosive placement, some of the models within the explosion area can remain in tact, and it would be THOSE models that were being registered as the installation.

Example: I destroyed a building and the installation marker in debug didn't go away. The building comprised of a small room (where the whiteboard and IED table was) and had a separate staircase that led to a second floor.

When I blew up the room with the whiteboard and IED table, that part of the building became destroyed. However, the staircase area that led to the next floor didn't fall down, but the building definitely LOOKED destroyed to the naked eye.

Using cursortarget setdamage 1, I noticed this staircase area has its own model. Once I blew THAT up, the installation finally disappeared.

Thanks Highhead. Closing now. Will reopen if I discover there is an issue. But I think we're good.
